### PR TITLE
IntelFsp2Pkg : Add FSP-I arch config PPI

### PR DESCRIPTION
**Pull request by hongbin123**
- [ ] Breaking change?
  - **Breaking change** - Does this PR cause a break in build or boot behavior?
  - Examples: Does it add a new library class or move a module to a different repo.
- [ ] Impacts security?
  - **Security** - Does this PR have a direct security impact?
  - Examples: Crypto algorithm change or buffer overflow fix.
- [ ] Includes tests?
  - **Tests** - Does this PR include any explicit test code?
  - Examples: Unit tests or integration tests.

### CODE REVIEW

**Review by niruiyu — CHANGES_REQUESTED** [review]

File: IntelFsp2Pkg/IntelFsp2Pkg.dec [hunk 1]
```diff
@@ -67,6 +67,10 @@
   #
   gEdkiiPeiVariablePpiGuid              = { 0xe7b2cd04, 0x4b14, 0x44c2, {0xb7, 0x48, 0xce, 0xaf, 0x2b, 0x66, 0x4a, 0xb0}}
 
+  #
+  # PPI for Architectural configuration data for FSP-I
+  #
```
Comment (jiaxinwu) [inline IntelFsp2Pkg/IntelFsp2Pkg.dec:72]: change Copyright (c) 2014 - 2024

File: IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+///
+/// Global ID for the FSPI_ARCH_CONFIG_PPI.
+///
+#define FSPI_ARCH_CONFIG_GUID \
+  { \
+    0x15735ef9, 0x84ac, 0x4e34, { 0x98, 0x86, 0x56, 0xee, 0xcf, 0x73, 0xec, 0x0f } \
+  }
+
+///
+/// This PPI provides FSP-M Arch Config PPI.
```
Comment (jiaxinwu) [inline IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h:24]: FSP-I Arch

File: IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  /// Revision of the structure is 1 for this version of the specification.
+  ///
+  UINT8                   Revision;
+  UINT8                   Reserved[3];
+  ///
+  /// Length of the structure in bytes. The current value for this field is 64.
+  ///
+  UINT32                  Length;
+  ///
+  /// The physical memory-mapped base address of the bootloader SMM firmware volume (FV).
```
Comment (jiaxinwu) [inline IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h:37]: bootloader MM firmware volume

File: IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  ///
+  EFI_PHYSICAL_ADDRESS    BootloaderSmmFvBaseAddress;
+  ///
+  /// The length in bytes of the bootloader SMM firmware volume (FV).
```
Comment (jiaxinwu) [inline IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h:41]: bootloader MM

File: IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  ///
+  UINT64                  BootloaderSmmFvLength;
+  ///
+  /// The physical memory-mapped base address of the bootloader SMM FV context data.
+  /// This data is provided to bootloader SMM drivers through a HOB by the FSP MM Foundation.
```
Comment (jiaxinwu) [inline IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h:46]: FSP I Foundation.

File: IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  UINT8                   Revision;
+  UINT8                   Reserved[3];
+  ///
+  /// Length of the structure in bytes. The current value for this field is 64.
+  ///
+  UINT32                  Length;
+  ///
+  /// The physical memory-mapped base address of the bootloader SMM firmware volume (FV).
+  ///
+  EFI_PHYSICAL_ADDRESS    BootloaderSmmFvBaseAddress;
```
[the remark this answers is not in the document]
Comment (hongbin123) [reply IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h]: updated

File: IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  ///
+  /// The length in bytes of the bootloader SMM firmware volume (FV).
+  ///
+  UINT64                  BootloaderSmmFvLength;
```
[the remark this answers is not in the document]
Comment (hongbin123) [reply IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h]: updated

File: IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+  ///
+  /// The physical memory-mapped base address of the bootloader SMM FV context data.
+  /// This data is provided to bootloader SMM drivers through a HOB by the FSP MM Foundation.
+  ///
+  EFI_PHYSICAL_ADDRESS    BootloaderSmmFvContextData;
```
[the remark this answers is not in the document]
Comment (hongbin123) [reply IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h]: updated

File: IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+  ///
+  /// The length in bytes of the bootloader SMM FV context data.
+  /// This data is provided to bootloader SMM drivers through a HOB by the FSP MM Foundation.
+  ///
+  UINT64                  BootloaderSmmFvContextDataLength;
```
[the remark this answers is not in the document]
Comment (hongbin123) [reply IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h]: updated

File: IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+  UINT8                   Reserved1[30];
```
[the remark this answers is not in the document]
Comment (hongbin123) [reply IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h]: ok

File: IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h (hunk 3 above)
Comment (hongbin123) [reply IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h:33]: if it is ok that line 39 name is different with description

File: IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+  ///
+  UINT64                  BootloaderSmmFvLength;
+  ///
+  /// The physical memory-mapped base address of the bootloader SMM FV context data.
+  /// This data is provided to bootloader SMM drivers through a HOB by the FSP MM Foundation.
+  ///
+  EFI_PHYSICAL_ADDRESS    BootloaderSmmFvContextData;
+  ///
+  /// The length in bytes of the bootloader SMM FV context data.
+  /// This data is provided to bootloader SMM drivers through a HOB by the FSP MM Foundation.
```
[the remark this answers is not in the document]
Comment (hongbin123) [reply IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h]: updated

File: IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h (hunk 3 above)
Comment (hongbin123) [reply IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h:33]: keep SMM as defined in FSPI ARCH UPD

File: IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h (hunk 5 above)
Comment (hongbin123) [reply IntelFsp2Pkg/Include/Ppi/FspiArchConfigPpi.h]: updated

**Review by jiaxinwu — APPROVED** [review]

**Review by niruiyu — APPROVED** [review]

**Pull request merged**